python_src/src/lamp_py/ingestion/pipeline.py

```diff
@@ -71,23 +83,67 @@ def ingest(metadata_queue: Queue) -> None:
         file_prefix=DEFAULT_S3_PREFIX,
     )
 
-    ingest_files(files, metadata_queue)
+    processed_files = ingest_files(files, metadata_queue)
+
+    process_logger.log_complete()
+
+    return processed_files
+
+
+def wait_for_deletion(filenames: List[str]) -> None:
+    """
+    wait for all of the files in the filenames list to be removed from the
+    incoming bucket. log how many files are still
+    """
```

Comment:
new function that intersects files that should have been moved with the files still in the incoming bucket.

it waits until the intersection is empty or a timeout (which is a function of the number of files) is reached.

it logs files that aren't moved correctly.